### PR TITLE
Port april point fixes

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
@joyceerhl your fix for backgrounds is in the renderer extension. Can't quite recall if this will get picked up for a point release. Checking now.

**Review by DavidKutu — APPROVED** [review]

**Comment by joyceerhl** [comment]
@IanMatthewHuff, sorry, I just saw this tag. The renderers fix doesn't seem to have been picked up. I think that's because we need to reupload the VSIX to https://pvsc.blob.core.windows.net/extension-builds-jupyter-public/ms-notebook-renderers.vsix. That VSIX gets downloaded when building the VSIX on CI and bundled with the rest of our renderers: https://github.com/microsoft/vscode-jupyter/blob/cabca7aa4cd350b1733ae74dffaa6ccb82d9daa5/build/ci/downloadRenderer.js#L36 

I've upload a new build of VSIX and triggered a new CI build on this PR. I'll test the output fix in the built VSIX when it's ready.

**Comment by joyceerhl** [comment]
Looks good now. Installed https://github.com/microsoft/vscode-jupyter/suites/2453762623/artifacts/52663220 in Stable VS Code and ran the following plot:

```python
import matplotlib
import matplotlib.pyplot as plt
import numpy as np

t = np.arange(0.0, 2.0, 0.01)
s = 1 + np.sin(2 * np.pi  * t)

fig, ax = plt.subplots()
ax.plot(t, s)

ax.set(xlabel='time (s)', ylabel='voltage (mV)',
    title='About as simple as it gets, folks')

ax.grid()

fig.savefig('test.png')

plt.show()
```

image

**Comment by IanMatthewHuff** [comment]
One more test still not passing, due to insiders / stable difference I think. One more try then we'll just manually test that at bug bash.

File: news/2 Fixes/5470.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Invalidate cached interpreters when Python extension active interpreter changes.
```
Comment (DavidKutu) [inline news/2 Fixes/5470.md:1]: This needs to be deleted @joyceerhl

Comment (joyceerhl) [reply news/2 Fixes/5470.md:1]: Oh yes, thanks

**Comment by joyceerhl** [comment]
@IanMatthewHuff, looks like @DavidKutu tracked down the failing tests. Are we good to merge this for the point release?

**Pull request merged**